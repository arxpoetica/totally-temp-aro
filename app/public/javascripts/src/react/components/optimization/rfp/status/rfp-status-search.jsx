import React, { Component } from 'react'
import { connect } from 'react-redux'
import { PropTypes } from 'prop-types'
import RfpActions from '../rfp-actions'

export class RfpStatusSearch extends Component {
  constructor (props) {
    super(props)
    this.state = {
      searchTerm: ''
    }
  }

  render () {
    return <div className='input-group mb-3'>
      <input
        id='txtRfpPlanSearch'
        type='text'
        className='form-control'
        placeholder='Enter search term...'
        aria-label='Enter search term"'
        value={this.state.searchTerm}
        onChange={event => this.onSearchTermChanged(event.target.value)}
        onKeyPress={event => this.onSearchKeyPress(event)}
        disabled={this.props.isLoadingRfpPlans}
      />
      <div className='input-group-append'>
        <button
          id='btnRfpPlanSearch'
          className='btn btn-primary'
          onClick={event => this.doSearch()}
          disabled={this.props.isLoadingRfpPlans}
        >
          <i className='fa fa-search' /> Search
        </button>
      </div>
    </div>
  }

  onSearchTermChanged (newSearchTerm) {
    this.setState({
      searchTerm: newSearchTerm
    })
  }

  onSearchKeyPress (event) {
    if (event.key === 'Enter') {
      this.doSearch()
    }
  }

  doSearch () {
    this.props.loadRfpPlans(this.props.userId, this.state.searchTerm)
  }
}

RfpStatusSearch.propTypes = {
  isLoadingRfpPlans: PropTypes.bool,
  userId: PropTypes.number
}

const mapStateToProps = state => ({
  isLoadingRfpPlans: state.optimization.rfp.isLoadingRfpPlans,
  userId: state.user.loggedInUser && state.user.loggedInUser.id
})

const mapDispatchToProps = dispatch => ({
  loadRfpPlans: (userId, searchTerm) => dispatch(RfpActions.loadRfpPlans(userId, searchTerm))
})

const RfpStatusSearchComponent = connect(mapStateToProps, mapDispatchToProps)(RfpStatusSearch)
export default RfpStatusSearchComponent
