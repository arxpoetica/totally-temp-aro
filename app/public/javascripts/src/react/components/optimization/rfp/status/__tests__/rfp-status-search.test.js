/* global jest test expect */
import React from 'react'
import { shallow } from 'enzyme'
import { RfpStatusSearch } from '../rfp-status-search.jsx'

// -----------------------------------------------------------------------------
test('When system is loading RFP plans', () => {
  const component = shallow(
    <RfpStatusSearch
      isLoadingRfpPlans
    />
  )
  expect(component).toMatchSnapshot()
})

// -----------------------------------------------------------------------------
test('When system is not loading RFP plans', () => {
  const component = shallow(
    <RfpStatusSearch
      isLoadingRfpPlans={false}
    />
  )
  expect(component).toMatchSnapshot()
})

// -----------------------------------------------------------------------------
test('When user clicks on button to search', () => {
  const userId = 42
  const mockLoadRfpPlans = jest.fn()
  const component = shallow(
    <RfpStatusSearch
      isLoadingRfpPlans={false}
      userId={userId}
      loadRfpPlans={mockLoadRfpPlans}
    />
  )
  const searchTerm = 'Rfp Plan'
  component.find('#txtRfpPlanSearch').simulate('change', { target: { value: searchTerm } })
  component.find('#btnRfpPlanSearch').simulate('click')
  expect(mockLoadRfpPlans).toHaveBeenCalledWith(userId, searchTerm)
})

// -----------------------------------------------------------------------------
test('When user presses enter key to search', () => {
  const userId = 42
  const mockLoadRfpPlans = jest.fn()
  const component = shallow(
    <RfpStatusSearch
      isLoadingRfpPlans={false}
      userId={userId}
      loadRfpPlans={mockLoadRfpPlans}
    />
  )
  const searchTerm = 'Rfp Plan'
  component.find('#txtRfpPlanSearch').simulate('change', { target: { value: searchTerm } })
  component.find('#txtRfpPlanSearch').simulate('keyPress', { key: 'Enter' })
  expect(mockLoadRfpPlans).toHaveBeenCalledWith(userId, searchTerm)
})
