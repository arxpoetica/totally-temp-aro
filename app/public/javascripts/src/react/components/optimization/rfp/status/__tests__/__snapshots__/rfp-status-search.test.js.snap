// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When system is loading RFP plans 1`] = `
<div
  className="input-group mb-3"
>
  <input
    aria-label="Enter search term\\""
    className="form-control"
    disabled={true}
    id="txtRfpPlanSearch"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Enter search term..."
    type="text"
    value=""
  />
  <div
    className="input-group-append"
  >
    <button
      className="btn btn-primary"
      disabled={true}
      id="btnRfpPlanSearch"
      onClick={[Function]}
    >
      <i
        className="fa fa-search"
      />
       Search
    </button>
  </div>
</div>
`;

exports[`When system is not loading RFP plans 1`] = `
<div
  className="input-group mb-3"
>
  <input
    aria-label="Enter search term\\""
    className="form-control"
    disabled={false}
    id="txtRfpPlanSearch"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Enter search term..."
    type="text"
    value=""
  />
  <div
    className="input-group-append"
  >
    <button
      className="btn btn-primary"
      disabled={false}
      id="btnRfpPlanSearch"
      onClick={[Function]}
    >
      <i
        className="fa fa-search"
      />
       Search
    </button>
  </div>
</div>
`;
